server/balancer_worker.go

```diff
@@ -113,6 +126,7 @@ func (bw *balancerWorker) removeBalanceOperator(regionID uint64) {
 	defer bw.Unlock()
 
 	delete(bw.balanceOperators, regionID)
+	bw.regionCache.delete(regionID)
```

Comment:
Maybe not, if the balancer is removed, actually it is not succeeded.
Then we should not use it as a filter because the last successful operator for this region may be in a long time before.